### ISSUE TITLE
Test Suite: accept more than one PSA_STATUS result from data files.

### ISSUE THREAD

**Issue by oberon-sk** [ARMmbed/mbedtls#7457]
### Suggested enhancement

The test function `derive_key()` in [test_suite_psa_crypto.function](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_psa_crypto.function#L9171) should accept more the one expected result of `PSA_STATUS`-codes.

### Justification

The test suite data file [test_suite_psa_crypto.data](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_psa_crypto.data) contains thirty tests with the comment `The spec allows either INVALID_ARGUMENT or NOT_SUPPORTED`.

Example: The test function `derive_key()` in [test_suite_psa_crypto.function](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_psa_crypto.function#L9171) expects a `PSA_ERROR_INVALID_ARGUMENT` result for an ECC key with an invalid  type-bits combination, see tests `PSA key derivation: bits=7 invalid for ECC MONTGOMERY` in [test_suite_psa_crypto.data](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_psa_crypto.data#L6371). However, a `PSA_ERROR_NOT_SUPPORTED` error is valid as well if the curve is not supported by the implementation at all.

**Comment by gilles-peskine-arm** [comment]
I think we should do what we're already doing in a few other places (e.g. see `test_equal_status` in `test_suite_psa_crypto_op_fail.function`), which is that the code in the `.function` file should accept `PSA_ERROR_INVALID_ARGUMENT` and `PSA_ERROR_NOT_SUPPORTED` as equivalent.

(In hindsight, I wouldn't have defined a `NOT_SUPPORTED` error, or at least not used it in this way. It has advantages, but it's caused so much friction like this that I don't think it was worth it.)

**Comment by oberon-sk** [comment]
Agreed. Since currently, `PSA_ERROR_NOT_SUPPORTED` and `PSA_ERROR_INVALID_ARGUMENT` must both be accepted in many cases, what should implementers choose? A simple rule for recommendation could be:

- use `PSA_ERROR_NOT_SUPPORTED` whenever a value of an extensible set is not handled, irrespective of the reason why,
- use `PSA_ERROR_INVALID_ARGUMENT` only for illegal arguments like null pointers or illegal sizes.

What do you think?
